Branch-SDK/src/io/branch/referral/ServerResponse.java

```diff
@@ -135,4 +140,28 @@ public JSONArray getArray() {
 	public BranchLinkData getLinkData() {
 		return linkData_;
 	}
+
+	/**
+	 * <p>Sets the request object caused this response, as a generic {@link Object}
+	 * instance. This object can be type-cast by other methods within this class:</p>
+	 *
+	 * @param request Generic {@link Object} instance containing the request data caused this response.
+	 */
+	public void setRequestObject(Object request) {
+		this.requestObject_ = request;
+	}
+
+	/**
+	 * <p>Checks whether the request params cause this response is an instance of a
+	 * {@link JSONObject} object, and if so type-casts it to the corresponding class type.</p>
+	 *
+	 * @return A {@link JSONObject} containing the post params sent with corresponding  server request, or a null
+	 * value if the post params is not of the {@link JSONObject} type.
+	 */
+	public JSONObject getRequestObject() {
```

Comment:
I have  a ticket for creating many thread.I am looking forward to create  an AsyncTasks for requests so that we can control the thread pool size. That way I can confine the callbacks and the requests params a single object